gen/build_deploy/bash.py

```diff
@@ -361,6 +361,48 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
     fi
 }
 
+function d_type_enabled_if_xfs()
+{
+    # Return 1 if $1 is a directory on XFS volume with ftype ! = 1
+    # otherwise return 0
+    DIRNAME="$1"
+
+    RC=0
+    if [[ ! -d "$DIRNAME" ]]; then
+        return $RC
```

Comment:
This short-circuit return creates the opportunity for a client of this function to pass a path that doesn't exist and then mistakenly believe that it's on a correctly mounted volume. I'd expect this function to either 1) return an error if the directory doesn't exist, or 2) walk up the directory tree until it finds a parent dir that exists and then check it. Option 1 sounds appropriate for a bash script's helper function, since it's simpler and more straightforward.